File: core/src/rustyice/core/GeneralSettings.java
package rustyice.core;

import com.badlogic.gdx.Preferences;

public class GeneralSettings {
    public static final String APPID = "rustyice.";

    private Preferences preferences;
    private int width;
    private int height;
    private boolean fullscreen;
    private float soundVolume;
    private float musicVolume;
    private boolean vsync;

    public GeneralSettings(Preferences preferences){
        this.preferences = preferences;

        width = preferences.getInteger(APPID + "width", 800);
        height = preferences.getInteger(APPID + "height", 640);
        fullscreen = preferences.getBoolean(APPID + "fullscreen", false);
        soundVolume = preferences.getFloat(APPID + "soundVolume", 1);
        musicVolume = preferences.getFloat(APPID + "musicVolume", 1);
        vsync = preferences.getBoolean(APPID + "vsync", true);
    }

    public void save(){
        preferences.flush();
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        this.width = width;
        preferences.putInteger(APPID + "width", width);
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        this.height = height;
        preferences.putInteger(APPID + "height", height);
    }

    public boolean isFullscreen() {
        return fullscreen;
    }

    public void setFullscreen(boolean fullscreen) {
        this.fullscreen = fullscreen;
        preferences.putBoolean(APPID + "fullscreen", fullscreen);
    }

    public float getSoundVolume() {
        return soundVolume;
    }

    public void setSoundVolume(float soundVolume) {
        this.soundVolume = soundVolume;
        preferences.putFloat(APPID + "sound", soundVolume);
    }

    public float getMusicVolume() {
        return musicVolume;
    }

    public void setMusicVolume(float musicVolume) {
        this.musicVolume = musicVolume;
        preferences.putFloat(APPID + "music", musicVolume);
    }

    public boolean isVsync() {
        return vsync;
    }

    public void setVSync(boolean vsync) {
        this.vsync = vsync;
        preferences.putBoolean(APPID + "vsync", vsync);
    }
}

File: core/src/rustyengine/GeneralSettings.kt
package rustyengine;

import com.badlogic.gdx.Preferences;

object GeneralSettings {
    private const val APPID = "rustyice"

    var preferences: Preferences? = null
        set(value) {
            field = value

            if(value != null) {
                width = value.getInteger("${APPID}.width", width)
                height = value.getInteger("${APPID}.height", height)
                fullscreen = value.getBoolean("${APPID}.fullscreen", fullscreen)
                soundVolume = value.getFloat("${APPID}.soundVolume", soundVolume)
                musicVolume = value.getFloat("${APPID}.musicVolume", musicVolume)
                vsync = value.getBoolean("${APPID}.vsync", vsync)
            }
        }

    var width: Int = 800
        set(value) {
            field = value
            preferences?.putInteger("${APPID}.width", value)
        }

    var height: Int = 640
        set(value) {
            field = value
            preferences?.putInteger("${APPID}.width", value)
        }

    var soundVolume: Float = 1f
        set(value) {
            field = value
            preferences?.putFloat("${APPID}.soundVolume", value)
        }

    var musicVolume: Float = 1f
        set(value) {
            field = value
            preferences?.putFloat("${APPID}.musicVolume", value)
        }

    var vsync: Boolean = true
        set(value) {
            field = value
            preferences?.putBoolean("${APPID}.vsync", value)
        }

    var fullscreen: Boolean = false
        set(value) {
            field = value
            preferences?.putBoolean("${APPID}.fullscreen", value)
        }

    fun save(){
        preferences?.flush()
    }
}
